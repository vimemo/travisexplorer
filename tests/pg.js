import Pg, {ensureDbExists} from '../lib/pg'

const DB_NAME = 'testdb'
const TABLE_NAME = 'testreplica'
const DB_URL = `${process.env.TEST_PG_URL}/${DB_NAME}`

let pgdb, pgtable

describe('pg', () => {

  test('invalid constructor', () => {
    expect(() => new Pg()).toThrowError(/Missing Parameter/)
  })

  describe('functional', () => {
    beforeEach(async () => {
      await ensureDbExists(DB_URL)
      pgdb = new Pg(DB_URL)
      await pgdb.dropTableIfExists(TABLE_NAME)
      pgtable = await pgdb.createTableIfNotExists(TABLE_NAME, table => {
        table.increments()
        table.string('name')
        table.string('description')
      })
    })

    afterEach(async () => { await pgdb.db.destroy() })

    test('insert, get', async () => {
      await pgtable.insert([
        {name: 'A', description: 'AAA'},
        {name: 'B', description: 'BBB'},
        {name: 'C', description: 'CCC'}
      ])
      const rows = await pgtable.rows()
      expect(rows.length).toBe(3)
      expect(rows).toMatchSnapshot()
    })

    test('create table if not exists', async () => {
      const sametable = await pgdb.createTableIfNotExists(TABLE_NAME)
      expect(sametable.name).toBe(TABLE_NAME)
    })
  })
})
