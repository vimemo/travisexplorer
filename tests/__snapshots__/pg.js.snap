// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pg functional insert, get 1`] = `
Array [
  anonymous {
    "description": "AAA",
    "id": 1,
    "name": "A",
  },
  anonymous {
    "description": "BBB",
    "id": 2,
    "name": "B",
  },
  anonymous {
    "description": "CCC",
    "id": 3,
    "name": "C",
  },
]
`;
